File: src/commands/workspace.ts
import { queriesInit } from '@/commands/use'

type WorkspaceSchema = {
  path: string
  auto_add_new_project: boolean
}

export const { useGetWorkspaceQuery, useGetWorkspaceLazy } = queriesInit<{}, WorkspaceSchema>('get_workspace')

export const { useSetWorkspaceQuery, useSetWorkspaceLazy } = queriesInit<{ data: WorkspaceSchema }, {}>('set_workspace')

File: src-tauri/command/workspace.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct WorkspaceSchema {
    path: String,
    auto_add_new_project: bool,
}

#[tauri::command]
pub fn get_workspace() -> WorkspaceSchema {
    WorkspaceSchema { path: "".to_string(), auto_add_new_project: true }
}

#[tauri::command]
pub fn put_workspace(data: WorkspaceSchema) {
    println!("{:?}", data);
}
